option.go

```diff
@@ -119,6 +119,18 @@ func EqualFuncs(b bool) Option {
 	}}
 }
 
+// ZeroFields copies values and sets the specified fields to their zero value.
+func ZeroFields[T any](fields ...string) Option {
+	return Transform(func(v T) any {
+		rv := reflect.ValueOf(&v)
+		for _, name := range fields {
+			fv := rv.Elem().FieldByName(name)
```

Comment:
Consider calling Elem outside the loop. I dunno if that matters for performance but it couldn't hurt. (`ValueOf(&foo).Elem()` is a bit of an idiom to get a settable Value.)